### DISCUSSION TITLE
Table Action to View File Upload Directly

### DISCUSSION THREAD

**Discussion by ExpertMediaDesign** [filamentphp/filament#6660] [Help] [answered]
I have an idea working below, though I'd love to hear if there's a more "Filament-y" solution.

We have a Document Resource that has a FileUpload field (named 'file_path').
Our goal is to create a table action that is a link directly to the file (instead of the standard View action to the View page of the Resource).

Currently I can manually set the link this way:
```
Action::make('viewfile')->url(function (Document $record) {
    return '/storage/' . $record->file_path; 
 })
```
Though that feels clunky. Is there a better way to do this?
Appreciate any help. Thank you.

**Comment by pboivin** [comment, marked as answer]
Maybe this : `return Storage::url($record->file_path);`... it should add the '/storage/' prefix for you.

**Reply by ExpertMediaDesign** [reply]
Appreciate it! Definitely helps. Guess I assumed there might be a Filament or FilePond method for this, but that feels good for me.
